crates/pessimistic-proof/src/local_exit_tree/mod.rs

```diff
@@ -69,12 +71,10 @@ where
     }
 
     /// Appends a leaf to the tree.
-    pub fn add_leaf(&mut self, leaf: H::Digest) {
-        assert!(
-            (self.leaf_count as usize) < (1usize << TREE_DEPTH) - 1,
-            "Can have at most {} leaves.",
-            (1usize << TREE_DEPTH) - 1
-        );
+    pub fn add_leaf(&mut self, leaf: H::Digest) -> Result<u32, LocalExitTreeError> {
+        if self.leaf_count as u64 >> TREE_DEPTH != 0 {
+            return Err(LocalExitTreeError::LeafIndexOverflow);
+        }
```

Comment:
nit: Not very important but this checks that `self.leaf_count < 2^TREE_DEPTH`, when it should really check that `self.leaf_count < 2^TREE_DEPTH - 1` (the LET implementation doesn't support having `2^TREE_DEPTH - 1` leaves).